### PR TITLE
mark various 'partial constructors' as deprecated

### PR DESCRIPTION
**Pull request by kroening**
This adds a comment to various partial constructors in util/std_code.h.
The next step will be to make them generate a warning before they are removed.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 13fccfe5866e8aea1b113bfd25205a23bd9f1497).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77608381

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 54285dd95d69b0511f014a07d81222173048522a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77623405

File: src/util/std_code.h [hunk 1]
```diff
@@ -533,11 +538,19 @@ inline code_ifthenelset &to_code_ifthenelse(codet &code)
...
   code_switcht():codet(ID_switch)
   {
     operands().resize(2);
   }
 
+  code_switcht(const exprt &_value, const codet &_body):codet(ID_switch)
+  {
+    operands().resize(2);
+    value()=_value;
+    body()=_body;
```
Comment (tautschnig) [inline src/util/std_code.h]: Nit pick: clang-format likes a space around `=`

File: src/util/std_code.h [hunk 2]
```diff
@@ -533,11 +538,19 @@ inline code_ifthenelset &to_code_ifthenelse(codet &code)
 class code_switcht:public codet
 {
 public:
+  // deprecated, will go away
   code_switcht():codet(ID_switch)
   {
     operands().resize(2);
   }
 
+  code_switcht(const exprt &_value, const codet &_body):codet(ID_switch)
```
Comment (tautschnig) [inline src/util/std_code.h]: Nit pick: clang-format spacing around `:`

File: src/util/std_code.h [hunk 3]
```diff
@@ -21,6 +21,7 @@ Author: Daniel Kroening, kroening@kroening.com
 class codet:public exprt
 {
 public:
+  // deprecated, will go away
```
Comment (tautschnig) [inline src/util/std_code.h]: I'd strongly encourage using the tools from `deprecated.h`

File: src/goto-programs/goto_convert_side_effect.cpp [hunk 4]
```diff
@@ -440,7 +438,7 @@ void goto_convertt::remove_cpp_new(
   decl.add_source_location()=new_symbol.location;
   convert_decl(decl, dest, ID_cpp);
 
-  call=code_assignt(new_symbol.symbol_expr(), expr);
+  auto call=code_assignt(new_symbol.symbol_expr(), expr);
```
Comment (tautschnig) [inline src/goto-programs/goto_convert_side_effect.cpp]: I believe this should be `const code_assignt call(new_symbol.symbol_expr(), expr);`

File: src/ansi-c/c_typecheck_expr.cpp [hunk 5]
```diff
@@ -461,14 +461,16 @@ void c_typecheck_baset::typecheck_expr_builtin_va_arg(exprt &expr)
 
   implicit_typecast(arg, pointer_type(void_type()));
 
+  auto function=symbol_exprt(ID_gcc_builtin_va_arg, new_type);
```
Comment (tautschnig) [inline src/ansi-c/c_typecheck_expr.cpp]: `symbol_exprt function(...`

**Comment by kroening** [comment]
All done

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 081f743cde97c682356801bd726696ec71f31c50).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78280986

File: jbmc/src/java_bytecode/java_bytecode_instrument.cpp [hunk 6]
```diff
@@ -604,13 +604,14 @@ void java_bytecode_instrument_uncaught_exceptions(
   const source_locationt &source_location)
 {
   // check that there is no uncaught exception
-  code_assertt assert_no_exception;
-  assert_no_exception.assertion() = equal_exprt(
+  code_assertt assert_no_exception(equal_exprt(
```
Comment (peterschrammel) [inline jbmc/src/java_bytecode/java_bytecode_instrument.cpp:607]: A last nitpick from clang-format: https://travis-ci.org/diffblue/cbmc/jobs/400793974#L505

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**